### PR TITLE
Refactor item logistics for future LIMO work

### PR DESCRIPTION
**Pull request by ziadoz**
- Refactored out item logistics XML block into a new `Item\ItemLogistics` class.
- Added a new `RenderableInterface` so it's clear which objects can be rendered.
- Removed `$feed->getXml()` which is a public alias to the private `$feed->render()` and just made that public for consistency.
- Minor test code tweaks.

### CODE REVIEW

**Comment by gabriel403** [comment]
:+1: version 3.0 here we come!

File: src/Item.php [hunk 1]
```diff
@@ -567,133 +569,15 @@ public function setAssets(array $urls, $urlPrefix = '')
...
-     * Set the item logistics.
-     * Corresponding attributes will also be added to the product attributes group too.
+     * Set the item logistics elemnt.
+     * Note: This will also add corresponding attributes to the product attributes group.
      *
-     * @param array $itemLogistics (varidiac parameters in order 'legacyDistributorId', 'mdsfamId', 'vendorStockId', 'unitCost')
+     * @param ItemLogistics $itemLogistics
      */
-    public function setItemLogistics(...$itemLogistics)
+    public function setItemLogistics(ItemLogistics $itemLogistics)
```
Comment (gabriel403) [inline src/Item.php]: hmm can we type on renderable interface instead?

Comment (gabriel403) [inline src/Item.php]: Actually I don't think we can, can we

Comment (ziadoz) [inline src/Item.php]: I did think about doing that, but whatever gets passed in needs to have the `getProductAttributes()` method defined on it, as some of the item logistic parameters also get added to `<ProductAttributes>` too. If we removed that behaviour and did it explicitly in our code then we could probably just typehint `RenderableInterface`.

**Pull request merged**